File: binding/src/libqstringtokenizer.cpp
#include <QStringTokenizerBaseBase>
#include <qstringtokenizer.h>
#include "libqstringtokenizer.h"
#include "libqstringtokenizer.hxx"

QStringTokenizerBaseBase* QStringTokenizerBaseBase_new(const QStringTokenizerBaseBase* other) {
	 return new QStringTokenizerBaseBase(*other);
}

QStringTokenizerBaseBase* QStringTokenizerBaseBase_new2(const QStringTokenizerBaseBase* param1) {
	 return new QStringTokenizerBaseBase(*param1);
}

void QStringTokenizerBaseBase_CopyAssign(QStringTokenizerBaseBase* self, QStringTokenizerBaseBase* other) {
    *self = *other;
}

File: binding/include/libqstringtokenizer.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qstringtokenizerbasebase.html
pub const qstringtokenizerbasebase = struct {

    /// New constructs a new QStringTokenizerBaseBase object.
    pub fn New(other: ?*anyopaque) QtC.QStringTokenizerBaseBase {
        return qtc.QStringTokenizerBaseBase_new(@ptrCast(other));
    }


    /// New2 constructs a new QStringTokenizerBaseBase object.
    pub fn New2(param1: ?*anyopaque) QtC.QStringTokenizerBaseBase {
        return qtc.QStringTokenizerBaseBase_new2(@ptrCast(param1));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QStringTokenizerBaseBase_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

};
